stm32cube/stm32n6xx/CMakeLists.txt

```diff
@@ -0,0 +1,114 @@
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_GPIO drivers/src/stm32n6xx_hal_gpio.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_GPU2D drivers/src/stm32n6xx_hal_gpu2d.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_HASH drivers/src/stm32n6xx_hal_hash.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_HCD drivers/src/stm32n6xx_hal_hcd.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_I2C drivers/src/stm32n6xx_hal_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_I2C_EX drivers/src/stm32n6xx_hal_i2c_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_I2S drivers/src/stm32n6xx_hal_i2s.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_I2S_EX drivers/src/stm32n6xx_hal_i2s_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_I3C drivers/src/stm32n6xx_hal_i3c.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_ICACHE drivers/src/stm32n6xx_hal_icache.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_IRDA drivers/src/stm32n6xx_hal_irda.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_IWDG drivers/src/stm32n6xx_hal_iwdg.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_JPEG drivers/src/stm32n6xx_hal_jpeg.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_LPTIM drivers/src/stm32n6xx_hal_lptim.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_LTDC drivers/src/stm32n6xx_hal_ltdc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_LTDC_EX drivers/src/stm32n6xx_hal_ltdc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_MCE drivers/src/stm32n6xx_hal_mce.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_MDF drivers/src/stm32n6xx_hal_mdf.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_MDIOS drivers/src/stm32n6xx_hal_mdios.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_MMC drivers/src/stm32n6xx_hal_mmc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_MMC_EX drivers/src/stm32n6xx_hal_mmc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_NAND drivers/src/stm32n6xx_hal_nand.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_NOR drivers/src/stm32n6xx_hal_nor.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_PCD drivers/src/stm32n6xx_hal_pcd.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_PCD_EX drivers/src/stm32n6xx_hal_pcd_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_PKA drivers/src/stm32n6xx_hal_pka.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_PSSI drivers/src/stm32n6xx_hal_pssi.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_PWR drivers/src/stm32n6xx_hal_pwr.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_PWR_EX drivers/src/stm32n6xx_hal_pwr_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_RAMCFG drivers/src/stm32n6xx_hal_ramcfg.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_RCC drivers/src/stm32n6xx_hal_rcc.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_RCC_EX drivers/src/stm32n6xx_hal_rcc_ex.c)
+zephyr_library_sources_ifdef(CONFIG_USE_STM32_HAL_RIF drivers/src/stm32n6xx_hal_rif.c)
```

Comment:
RIF symbol is not defined yet on Zephyr repo side. To be added as part of https://github.com/zephyrproject-rtos/zephyr/pull/84128

Reply:
Symbols added in modules/Kconfig.stm32.
Also removed CONFIG_USE_STM32_HAL_RCC and CONFIG_USE_STM32_HAL_RCC_EX from this PR (RCC files are always included without conditions)